jetty9/src/com/thoughtworks/go/server/Jetty9Server.java

```diff
@@ -0,0 +1,242 @@
+import java.io.PrintWriter;
+import java.lang.management.ManagementFactory;
+
+public class Jetty9Server extends AppServer {
+    private Server server;
+    private WebAppContext webAppContext;
+    private static final Logger LOG = Logger.getLogger(Jetty9Server.class);
+    private GoCipherSuite goCipherSuite;
+
+    public Jetty9Server(SystemEnvironment systemEnvironment, String password, SSLSocketFactory sslSocketFactory) {
+        this(systemEnvironment, password, sslSocketFactory, new Server());
+    }
+
+    Jetty9Server(SystemEnvironment systemEnvironment, String password, SSLSocketFactory sslSocketFactory, Server server){
+        super(systemEnvironment, password, sslSocketFactory);
+        this.server = server;
+        goCipherSuite = new GoCipherSuite(sslSocketFactory);
+    }
+
+    @Override
+    public void configure() throws Exception {
+        server.addEventListener(mbeans());
+        server.addConnector(plainConnector());
+        server.addConnector(sslConnector());
+        HandlerCollection handlers = new HandlerCollection();
+        handlers.addHandler(welcomeFileHandler());
+        handlers.addHandler(legacyRequestHandler());
+        createWebAppContext();
+        addResourceHandler(handlers, webAppContext);
+        handlers.addHandler(webAppContext);
+        server.setHandler(handlers);
+        performCustomConfiguration();
+        server.setStopAtShutdown(true);
+    }
+
+    @Override
+    public void start() throws Exception {
+        server.start();
+    }
+
+    @Override
+    public void stop() throws Exception {
+        server.stop();
+    }
+
+    @Override
+    public void addExtraJarsToClasspath(String extraClasspath) {
+        webAppContext.setExtraClasspath(extraClasspath);
+    }
+
+    @Override
+    public void setCookieExpirePeriod(int cookieExpirePeriod) {
+        webAppContext.getSessionHandler().getSessionManager().getSessionCookieConfig().setMaxAge(cookieExpirePeriod);
+    }
+
+    @Override
+    public void setInitParameter(String name, String value) {
+        webAppContext.setInitParameter(name, value);
+    }
+
+    @Override
+    public void addStopServlet() {
+        ServletHolder holder = new ServletHolder();
+        holder.setServlet(new StopJettyFromLocalhostServlet(this));
+        webAppContext.addServlet(holder, "/jetty/stop");
+    }
+
+    @Override
+    public Throwable getUnavailableException() {
+        return webAppContext.getUnavailableException();
+    }
+
+    private MBeanContainer mbeans() {
+        MBeanServer platformMBeanServer = ManagementFactory.getPlatformMBeanServer();
+        MBeanContainer mbeans = new MBeanContainer(platformMBeanServer);
+        return mbeans;
+    }
+
+    ContextHandler welcomeFileHandler() {
+        return new GoServerWelcomeFileHandler();
+    }
+
+    Handler legacyRequestHandler() {
+        return new LegacyUrlRequestHandler();
+    }
+
+    static class LegacyUrlRequestHandler extends HandlerWrapper {
```

Comment:
This is the perfect time/place to get rid of this. Worst case, it is being used in some place we don't know of, and there will have to be a switch to Jetty6. I think it can't get better than this, to remove this cruft.